.github/workflows/test_integration.yaml

```diff
@@ -0,0 +1,88 @@
+name: "Integration Tests"
+
+on:
+  schedule:
+    - cron: "0 0 * * MON"
+  workflow_dispatch:
+
+env:
+  CLOUDFLARE_TOKEN: ${{ secrets.CLOUDFLARE_TOKEN }}
+
+jobs:
+  test-integration:
+    name: "Pytest Integration"
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      matrix:
+        provider:
+          - aws
+          - do
+          - gcp
+    steps:
+      - name: "Checkout Infrastructure"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Install Nebari
+        run: |
+          pip install .[dev]
+          conda install --quiet --yes conda-build
+
+      - name: Retrieve secret from Vault
+        uses: hashicorp/vault-action@v2.5.0
+        with:
+          method: jwt
+          url: "https://quansight-vault-public-vault-b2379fa7.d415e30e.z1.hashicorp.cloud:8200"
+          namespace: "admin/quansight"
+          role: "repository-nebari-dev-nebari-role"
+          secrets: |
```

Comment:
I open this PR to upgrade the permission for service accounts / OIDC for this repo: https://github.com/Quansight/quansight-cloud-infrastructure/pull/129

Reply:
This PR also grants access to the Cloudflare API token:

```
kv/data/repository/nebari-dev/nebari/cloudflare/internal-devops@quansight.com/nebari-dev token | CLOUDFLARE_TOKEN;

```

Reply:
Awesome, thank you!

Reply:
This doesn't seems to exist:

> kv/data/repository/nebari-dev/nebari/cloudflare/internal-devops@quansight.com/nebari-dev

https://github.com/nebari-dev/nebari/actions/runs/5750272204/job/15586702394

Reply:
Got it working, it was `nebari-dev-ci`